src/components/MegaButton/__tests__/MegaButton.spec.js

```diff
@@ -0,0 +1,125 @@
+import '@testing-library/jest-dom';
+import { render, fireEvent } from '@testing-library/vue';
+import MegaButton from '../MegaButton.vue';
+
+let label;
+let props;
+
+describe('Megabutton', () => {
+
+  describe('Megatext format', () => {
+
+    const megatextProps = {
+      id: 'megabutton',
+      modelValue: '',
+      name: 'Chonky Boi',
+      label: 'Megabutton',
+      value: 'bigOlButton',
+      disabled: false
+    };
+
+    it('disables the input when disabled prop is true', () => {
+      props = {
+        ...megatextProps,
+        disabled: true
+      };
+
+      const { getByLabelText } = render(MegaButton, {
+        props
+      });
+
+      const mega = getByLabelText(props.label);
+      expect(mega).toBeDisabled();
+    });
+
+    it('checks the input when the value matches the modelValue', () => {
+      const props = {
+        ...megatextProps,
+        modelValue: megatextProps.value
+      };
+      const { getByLabelText } = render(MegaButton, {
+        props
+      });
+
+      const mega = getByLabelText(props.label);
+      expect(mega).toBeChecked();
+    });
+
+    it('fires the input and click events when the input is clicked', async () => {
+      props = megatextProps;
+      const { getByLabelText, emitted } = render(MegaButton, {
+        props
+      });
+      const mega = getByLabelText(props.label);
+
+      await fireEvent.click(mega);
+      const emittedEvent = emitted();
+      expect(emittedEvent).toHaveProperty('click');
+      expect(emittedEvent).toHaveProperty('input');
+      expect(emittedEvent.input[0]).toEqual([props.value]);
+    });
+
+    describe('when disabled', () => {
+
+      describe('without disabled banner', () => {
+
+        beforeAll(() => {
+
+          props = {
+            ...megatextProps,
+            disabled: true
+          };
+
+          const { getByText } = render(MegaButton, {
+            props
+          });
+
+          label = getByText(props.label).closest('label');
```

Comment:
Same thing here, don't reuse the `label` variable, just create a new one that is scoped to this test.